### DISCUSSION TITLE
Property 'sessionClaims' does not exist on type 'Promise<ClerkMiddlewareAuthObject>'.ts(2339)

### DISCUSSION THREAD

**Discussion by adeyinka-dev** [clerk/javascript#4519] [Support request - DO NOT OPEN HERE]
Property 'sessionClaims' does not exist on type 'Promise<ClerkMiddlewareAuthObject>'.ts(2339) on Next Js v15

**Comment by adeyinka-dev** [comment]
import { clerkMiddleware, createRouteMatcher } from '@clerk/nextjs/server';

`export default clerkMiddleware(async (auth, req) => {
    const { sessionClaims } = await auth(); 
    const role = (sessionClaims?.metadata as { role?: string })?.role;

    for (const { matcher, allowedRoles } of matchers) {
        if (matcher(req) && !allowedRoles.includes(role!)) {
            return NextResponse.redirect(new URL(`/${role}`, req.url));
        }
    }
});`

**Reply by meifruit** [reply]
const authObject = await auth();
 const { sessionClaims } = authObject;

**Reply by MohammadAliiiii** [reply]
thanks @meifruit

**Comment by jescalan** [comment]
If this is still an issue, open it with [clerk support](https://clerk.com/contact/support) please!
